tensorflow/python/keras/layers/recurrent.py

```diff
@@ -1320,6 +1320,9 @@ def __init__(self,
     else:
       self._enable_caching_device = kwargs.pop('enable_caching_device', False)
     super(SimpleRNNCell, self).__init__(**kwargs)
+    if units < 0:
+      raise ValueError("Received a negative value for `units`, ",
```

Comment:
Please use the message in core.py, which contains the invalid value:

raise ValueError(f"Received an invalid value for `units`, expected
                            f"a positive integer, got {units}.")

Reply:
I think the param validation should happen first (before it even reach the super.__init__). This could slightly improve the performance, in case there is any heavy lifting work in base layer init.